core/src/ru/spbau/blackout/GameWorld.java

```diff
@@ -59,7 +65,76 @@ public GameWorld() {
         return gameObjects.iterator();
     }
 
-    public void update(float delta) {
+    @Override
+    public synchronized void inplaceSerialize(ObjectOutputStream out) throws IOException, ClassNotFoundException {
+        out.writeInt(gameObjects.size());
+
+        for (GameObject object : this) {
+            //System.out.println("some object des");
+            //object.inplaceSerialize(out);
+
+            Vector2 sending = object.getPosition();
+            out.writeFloat(sending.x);
+            out.writeFloat(sending.y);
+            if (object instanceof GameUnit) {
+                GameUnit unit = (GameUnit) object;
+                sending = unit.getSelfVelocity();
+                out.writeFloat(sending.x);
+                out.writeFloat(sending.y);
+            }
+
+            // ???? do not work
```

Comment:
If vector coordinates are read and written by different threads, these threads can observe no updates in case of no explicit synchronization (to oversimplify, threads are never forced to update their caches). Try declaring getPosition and setPosition synchronized.